js/trek.js

```diff
@@ -0,0 +1,120 @@
+// Wave 3: While viewing a single trip, you can reserve a spot
+// Use a form to submit your name to reserve a spot on the trip you are viewing.
+
+$(document).ready(() => {
+
+  $('.tripView').hide();
+  $('.reveal').hide();
+
+
+  baseURL = 'https://trektravel.herokuapp.com/trips'
+
+  let loadTrips = function loadTrips() {
+    $.get(baseURL, (response) => {
+      $('.trip-button').hide();
+      response.forEach(function(trip) {
+        let tripInfo = `<li class="small-12 large-12 columns"><h3 data-id=${trip.id}> ${trip.name}</li>`;
+
+        $('.triplist ul').append(tripInfo);
+        $('.reveal').hide();
+      });
+    })
+    .fail(function(response){
+      console.log(response);
+      $('#fail').html('<p>Request was unsuccessful</p>')
+    })
+    .always(function(){
+      console.log('always even if we have success or failure');
+    });
+  };
+
+
+  let loadTrip = function loadTrip(id) {
+    $.get((baseURL + '/' + id),
+    (response) => {
+      let tripInfo = `<section class="indTrip rows" id="${response.id}">
+      <h2 class="large-8 text-center large-centered columns"> ${response.name} </h2>
+      <p class="large-8 large-centered text-center columns"><strong> About:</strong> ${response.about} </p>
+      <p class="large-6 text-center columns"> Continent: ${response.continent} </p>
+      <p class="large-6 text-center columns"> Category: ${response.category} </p>
+      <p class="large-6 text-center columns"> Week(s): ${response.weeks} </p>
+      <p class="large-6 text-center columns"> Cost: $${response.cost} </p>
+      </section>`;
+
+      // let resButton = `<section class="res_modal"><p><button class="res_button small-12 small-centered large-12 large-centered text-center columns" data-open="reservation">Reserve a spot!</button></p></section>`
+
+      $('.triplist ul').html("");
+      $('.triplist').append(tripInfo);
+
+      // $('.tripView').show();
+      $('.reveal').show();
+    })
+    .fail(function(response){
+      console.log(response);
+      $('#fail').html('<p>Request was unsuccessful</p>')
+    })
+    .always(function(){
+      console.log('always even if we have success or failure');
+    });
```

Comment:
You probably don't need `always` callbacks on all of these. I think of `always` like the `ensure` clause of a `begin/rescue` block: every once in a very long while it's incredibly useful, and otherwise I forget it's even there.